src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts

```diff
@@ -177,6 +180,50 @@ export class DotEditContentHtmlService {
         this.currentAction = DotContentletAction.EDIT;
     }
 
+    /**
+     * Set the same height to containers in the same row
+     *
+     * @param {DotLayout} pageLayout
+     * @memberof DotEditContentHtmlService
+     */
+    setContaintersSameHeight(pageLayout: DotLayout): void {
+        const doc = this.getEditPageDocument();
+        const rowsMaxHeight: number[] = [];
+        let rowHeight: number;
+
+        const containersLayoutIds = pageLayout.body.rows.map((row: DotLayoutRow) => {
+            return row.columns.map((column: DotLayoutColumn) => {
+                return {
+                    identifier: column.containers[0].identifier,
+                    uuid: column.containers[0].uuid
+                };
+            });
+        });
+
+        const containerDomElements = containersLayoutIds.map((containerRow: Array<DotPageContainer>, index: number) => {
+            rowHeight = 0;
+            return containerRow.map((container: DotPageContainer) => {
+                const querySelector = [
+                    `div[data-dot-object="container"]`,
+                    `[data-dot-identifier="${container.identifier}"]`,
+                    `[data-dot-uuid="${container.uuid}"]`
+                ].join('');
+                const containerElement = doc.querySelector(querySelector);
+                containerElement.style.height = 'auto';
+                if (containerElement.offsetHeight > rowHeight) {
+                    rowsMaxHeight[index] = rowHeight = containerElement.offsetHeight;
+                }
+                return containerElement;
+            });
+        });
+
+        containerDomElements.map((containerRow: Array<HTMLElement>, index: number) => {
```

Comment:
Why not a forEach instead of map? also can be chaining to the top.

Reply:
https://codeburst.io/javascript-map-vs-foreach-f38111822c0f